src/core/document/json.pl

```diff
@@ -2563,9 +2589,31 @@
     ).
 
 insert_document(Transaction, Document, ID) :-
+    insert_document(Transaction, Document, false, ID).
+
+insert_document(Transaction, Document, JSON, ID) :-
```

Comment:
Using `JSON` as an argument name here is very confusing. Here, `JSON` is `true` or `false`, but elsewhere, `JSON` is mainly used for a dictionary.